### DISCUSSION TITLE
Update Form input after Livewire roundtrip

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#2202] [Help]
I have a form with multiple reactive inputs and some not. I would like a non reactive input to be calculated depending on other input fields. I tried to add closure to the default() function but it seems this function is called only when the form is built but not during Livewire roundtrip. Am I wrong ? How to handle that please ?

**Comment by Z3d0X** [comment]
You can set the value of the calculated input on `afterStateUpdated()` of the reactive inputs.
Here is an example:
```php
Forms\Components\TextInput::make('name')
    ->reactive()
    ->afterStateUpdated(function (?string $state, \Closure $set, ?Model $record) {
        if (!$record) {
            $set('slug', Str::slug($state));
        }
    }),

Forms\Components\TextInput::make('slug'),
```
